src/System.Management.Automation/engine/Modules/GetModuleCommand.cs

```diff
@@ -482,14 +472,120 @@ private void GetLoadedModules(string[] names, IDictionary<string, ModuleSpecific
         {
             var modulesToWrite = Context.Modules.GetModules(names, all);
 
-            foreach (
-                var moduleInfo in
-                    modulesToWrite.Where(moduleInfo => ModuleMatch(moduleInfo, moduleSpecTable, PSEdition))
-                )
+            foreach (PSModuleInfo moduleInfo in FilterModulesForEditionAndSpecification(modulesToWrite, moduleSpecTable))
             {
                 WriteObject(moduleInfo);
             }
         }
+
+        /// <summary>
+        /// Filter an enumeration of PowerShell modules based on the required PowerShell edition
+        /// and the module specification constraints set for each module (if any).
+        /// </summary>
+        /// <param name="modules">The modules to filter through.</param>
+        /// <param name="moduleSpecificationTable">Module constraints, keyed by module name, to filter modules of that name by.</param>
+        /// <returns>All modules from the original input that meet both any module edition and module specification constraints provided.</returns>
+        private IEnumerable<PSModuleInfo> FilterModulesForEditionAndSpecification(
+            IEnumerable<PSModuleInfo> modules,
+            IDictionary<string, ModuleSpecification> moduleSpecificationTable)
+        {
+// Edition check only applies to Windows
+// System32 module path
```

Comment:
It's better to put the comment right above `if (!SkipEditionCheck)`, like
```
     // Edition check only applies to Windows System32 module path
     if (!SkipEditionCheck)
```

Reply:
I included this comment style sort of as a suggestion. We have a number of `#if`s in the code where it's not clear why the behaviour is version-specific. You're very right that in this case it's not needed, but in a number of the FullCLR/CoreCLR and UNIX/Windows cases, as time goes on we'll probably want to remove those conditionals -- the problem being that we'll have to spend half a day understanding the historical context of why they went in in the first place. But I suppose it's a bit late now 😄

Reply:
Year ago we removed tons `#if`s after migrating to .Net Core 2.0.
> where it's not clear why the behaviour is version-specific

That's why we did not delete everything :smile:
In fact, we left only what we want to port ( since .Net Core 2.1, 3.0, 4.0, ... :-) )

Reply:
Yeah that's what I'm thinking — it would be much easier to clear out the platform-specific code as .NET Core matures with documentation devoted to why we made it platform-specific in the first place!